lib/src/base_request_builder.dart

```diff
@@ -16,4 +16,11 @@ abstract class BaseRequestBuilder {
 
   /// Url template to use to build the URL for the current request builder.
   String urlTemplate;
+
+  /// Clones the current request builder.
+  T clone();
+
+  /// Clones the current request builder using [clone] and sets the given
+  /// [template] as the url template.
+  T withUrl(String template) => this.clone()..urlTemplate = template;
```

Comment:
This method should also set the parameters with a single entry: key is going to be the raw url key you have in request information, value is going to be the value from the parameter

Reply:
I'm afraid I don't follow... Can you point me to some other implementations of this method?

Reply:
I believe what @baywet is saying is that the the `withUrl` should not be setting the template but should be set with an an actual and valid Url(the request will be sent as is with the Url without having to template matching). 
This is done by adding a value with a special key so that if its present, no matching is done and the request is made directly. This is done to override the behavior of matching in the event extra parameters are needed or a custom url is used. 
https://github.com/microsoft/kiota-abstractions-dotnet/blob/4bff08d4ff90b24c705f71ba4d3baf89a85f31c0/src/RequestInformation.cs#L82C52-L82C61

Reply:
I moved the `withUrl` method to an extension method since it only accesses public members. It in turn uses `clone`, so in the end the class extending `BaseRequestBuilder` only needs to implement a deep clone of all members.

Reply:
Thanks @ricardoboss